.ci_support/migrations/numpy2.yaml

```diff
@@ -0,0 +1,53 @@
+__migrator:
+  build_number: 1
+  kind: version
+  commit_message: |
+    Rebuild for numpy 2.0
```

Comment:
As we are now building against NumPy in Numcodecs, this adds the NumPy 2 migrator so that this package is properly migrated for NumPy 2 support